components/ConnectionSection/index.tsx

```diff
@@ -0,0 +1,114 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import ButtonWithImage from '../ButtonWithImage';
+import TooltipWithDialog from '../TooltipWithDialog';
+import { ModalOverlayComponent } from '../ModalOverlayComponent';
+import { DmpIcon } from '@/components/Icons';
+import styles from './connectionSection.module.scss';
+import connectionData from './connection-types.json';
+
+import {
+  PressEvent
+} from 'react-aria-components';
+
+interface ConnectionSectionProps {
+  type: string;
+  title: string;
+  content?: string;
+  btnUrl: string;
+  btnImageUrl?: string;
+  btnText: string;
+}
+
+interface AuthData {
+  id: string;
+  token: string;
+}
+
+type CloseFunction = () => void;
+type HandleDeleteFunction = (e: PressEvent, close: CloseFunction) => void;
+type ConnectionDataType = {
+  [key: string]: { tooltipText: string, content: string }
+}
+
+/* This builds each third-party section on the connections page. If we have auth data saved for the
+current third party, then we show a different view with a 'cancel' icon. Otherwise, we just show
+a button asking if they want to connect.*/
+const ConnectionSection = ({
+  type,
+  title,
+  content,
+  btnUrl,
+  btnImageUrl,
+  btnText,
+}: ConnectionSectionProps) => {
+
+  const [authData, setAuthData] = useState<AuthData | null>(null);
+
+  const handleDelete: HandleDeleteFunction = async (e, close) => {
+    try {
+      /*TODO: Eventually we will need to call an API to delete the relevant token from the database.
+      For now, just to see it in action, we're using localStorage*/
+      localStorage.removeItem(`connectionData${type}`);
+      setAuthData(null);
+      close();
+    } catch (error) {
+      console.error('An error occurred while deleting the item:', error);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Related to my precious comments. We should probably just implement the correct server-side urls now and avoid using client-side storage, even though it's just for testing. I'm cautious we forget to remove code like this later on.